### PR TITLE
Support old HartRAO hardware

### PR DESCRIPTION
**Pull request by andrerenard**

### CODE REVIEW

File: config/hartrao_reduced_links.yaml [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  gpu_output_split_buffer_3:
+    kotekan_buffer: standard
+  gpu_output_split_buffer_4:
+    kotekan_buffer: standard
+  gpu_output_split_buffer_5:
+    kotekan_buffer: standard
+#  gpu_output_split_buffer_6:
+#    kotekan_buffer: standard
+#  gpu_output_split_buffer_7:
+#    kotekan_buffer: standard
```
Comment (james-s-willis) [inline config/hartrao_reduced_links.yaml:111]: Same here and all following commented out regions for GPUs 6 and 7,

File: config/hartrao_reduced_links.yaml [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+    - gpu_output_split_buffer_1
+    - gpu_output_split_buffer_2
+    - gpu_output_split_buffer_3
+    - gpu_output_split_buffer_4
+    - gpu_output_split_buffer_5
+#    - gpu_output_split_buffer_6
+#    - gpu_output_split_buffer_7
+
+metadata:
+  #kotekan_stage: chimeMetadataDump
```
Comment (james-s-willis) [inline config/hartrao_reduced_links.yaml:243]: Is this debug?

File: lib/opencl/kernels/kv_corr_sm_legacy.cl [hunk 3]
```diff
@@ -0,0 +1,113 @@
+//#define SAMPLES_PER_DATA_SET
+//#define NUM_ELEMENTS
+//#define NUM_FREQS
+//#define BLOCK_SIZE
+//#define COARSE_BLOCK_SIZE
```
Comment (james-s-willis) [inline lib/opencl/kernels/kv_corr_sm_legacy.cl:5]: These are commented out here but are actually used in the code.

**Review by james-s-willis — CHANGES_REQUESTED** [review]
So there are 6 GPU buffers which get merged into 1 buffer and are only processed by 1 GPU?

**Comment by andrerenard** [comment]
> So there are 6 GPU buffers which get merged into 1 buffer and are only processed by 1 GPU?

Yes that's correct.  With the CHIME configurations we always have one stream of data with 1 GPU, but with most other configurations we take a few streams of data and process them with the same GPU.

File: config/hartrao_reduced_links.yaml [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+  lost_samples_buffer_3:
+    kotekan_buffer: standard
+  lost_samples_buffer_4:
+    kotekan_buffer: standard
+  lost_samples_buffer_5:
+    kotekan_buffer: standard
+#  lost_samples_buffer_6:
+#    kotekan_buffer: standard
+#  lost_samples_buffer_7:
+#    kotekan_buffer: standard
```
[the remark this answers is not in the document]
Comment (andrerenard) [reply config/hartrao_reduced_links.yaml:77]: The system at HartRAO currently doesn't have enough working sticks of memory in the system to process all 8 streams.  I've left those in to un-comment once that's fixed.

File: config/hartrao_reduced_links.yaml (hunk 2 above)
Comment (andrerenard) [reply config/hartrao_reduced_links.yaml:243]: Oh yeah I didn't even notice that line was there.

File: lib/opencl/kernels/kv_corr_sm_legacy.cl (hunk 3 above)
Comment (andrerenard) [reply lib/opencl/kernels/kv_corr_sm_legacy.cl:5]: Ah those are passed in at run-time.  I'll add a note to that effect.

**Review by james-s-willis — APPROVED** [review]
Looks good to me now!

**Comment by dcrichton** [comment]
Looks good to me too with the cinttypes includes. Do you think it's worth adding the `subset_active` stage parts of the config that we're currently running to the repo? Given that the number of active elements will vary at HartRAO over the next few months I don't think so.

**Comment by andrerenard** [comment]
Yeah I don't think we need the `subset_active` part if that's actively changing.  I'm going to merge this in now.

**Pull request merged**